### PR TITLE
[Doc] Document `charts` plugin

### LINKED ISSUE #2585: [Research] Document `charts` plugin
1. What are it's capabilities?
2. Which visualizations use these capabilities?
3. Is it worth enhancing/extending for improved visualization abstractions?

### PR DESCRIPTION
**Pull request by joshuarrrr**
### Description

* Update README with full capabilities
* Add `current_usage` doc to track current usage

The primary value in documenting usage is to identify UX gaps to close as part of https://github.com/opensearch-project/OpenSearch-Dashboards/issues/2667, https://github.com/opensearch-project/OpenSearch-Dashboards/issues/2507, and https://github.com/opensearch-project/OpenSearch-Dashboards/issues/2385

 
### Issues Resolved

fixes #2585 
 
### Check List
- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [x] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Review by kavilla — APPROVED** [review]
Awesome thx!

**Review by ananzh — DISMISSED** [review]

**Review by kavilla — APPROVED** [review]

File: src/plugins/charts/DEVELOPER_GUIDE.md [hunk 1]
```diff
@@ -0,0 +1,180 @@
+# Usage
```
Comment (ashwin-pc) [inline src/plugins/charts/DEVELOPER_GUIDE.md:1]: Yeah, i'm not convinced this is the right place to have this information. Seems like the forked docsite might be a better place, but not sure how maintained that is currently. 

Didnt others create similar docs on how other plugins work, e.g. How visualizations work using a flowchart. where did we share that?

One other alternative to pave the way could be to just have it here but call it something more appropriate like `Current_useage.md`. Developer guide is what threw me off the most. That usually means something else.

Comment (joshuarrrr) [reply src/plugins/charts/DEVELOPER_GUIDE.md:1]: OK, in this new revision, I've moved to the `docs` directory and renamed.

> Didnt others create similar docs on how other plugins work, e.g. How visualizations work using a flowchart. where did we share that?

Not that I know of? I think some information like that is simply in GitHub issues, such as https://github.com/opensearch-project/OpenSearch-Dashboards/issues/2187

Comment (ashwin-pc) [reply src/plugins/charts/DEVELOPER_GUIDE.md]: I like that! That seems like a better place for high level information like this for now while we decide on the best place to consolidate this.

**Review by ashwin-pc — APPROVED** [review]

**Review by abbyhu2000 — APPROVED** [review]

**Review by ananzh — APPROVED** [review]

**Pull request merged**